rllib/algorithms/algorithm.py

```diff
@@ -620,6 +622,26 @@ def setup(self, config: AlgorithmConfig) -> None:
                 )
                 self._evaluation_weights_seq_number = 0
 
+        self.evaluation_dataset = None
+        if (
+            self.evaluation_config.off_policy_estimation_methods
+            and not self.evaluation_config.ope_split_batch_by_episode
```

Comment:
if ope is specified but not split_batch_by_episode we fall back to the old behavior. i.e. use evaluation workers that each have a shard of the dataset. It is not clear how I could use ray dataset map_batches to do processing on an episode level. In the future we can do that consolidation but till then let's just go back to the old behavior.

Reply:
ok. but how do I tell this is the behavior here?
like we are gonna get to this if statement of ``off_policy_estimation_methods and not ope_split_batch_by_episode``, if it doesn't fit, where do we fall to? how do I know the evaluation workers are guaranteed to be there?
to be clear, I am not saying the behavior is bad, I am just saying maybe we can write the code in a way that makes this flow clearer. wdyt?

Reply:
yep. Let me explain what happens:
if this condition is true, you'll set self.evaluation_dataset to a dataset otherwise it's None. In `self.evaluate()` if `self.evaluation_dataset` is None you fall back on to the old behavior. which is using evaluation_workers. Now the question is have we actually created evaluation_workers when they where necessary? The answer is yes. It's in the setup() function where you call `self._should_create_evaluation_rollout_workers(self.evaluation_config)`. How do you think this should be re-written so that it is more clear.

Reply:
ok, fine for now.
personally I will feel a lot safer, if this whole thing is a single if...else..., then I don't all cases are covered.
right now creation of ``self.evalution_workers`` is conditioned on ``self._should_create_evaluation_rollout_workers()``, and creation of ``evaluation_dataset`` is conditioned on ``off_policy_estimation_methods and not ope_split_batch_by_episode``.
it's just hard to tell if these 2 are mutually exclusive or not. you get the idea.